### PR TITLE
Support for new webhook endpoints

### PR DESCRIPTION
**Pull request by svenbw**
The new API (v3) has a new format for the webhooks, they return an array of urls and the hook names are changed.
This PR keeps the old methods to allow access to the "legacy webhooks" and adds new functions to the new webhooks:
- `getClickedUrls()`
- `getComplainedUrls()`
- `getDeliveredUrls()`
- `getOpenedUrls()`
- `getPermanentFailUrls()`
- `getTemporaryFailUrls()`
- `getUnsubscribeUrls()`

These new functions return an array.

### CODE REVIEW

**Comment by svenbw** [comment]
Fixes #599 Function `getWebhookUrl()` was renamed to `getWebhookUrls()` and returns an array.
Note that there is no support for the legacy webhooks in the show functions; this endpoint seems to be removed.

File: src/Model/Webhook/ShowResponse.php [hunk 1]
```diff
@@ -33,8 +33,8 @@ public static function create(array $data): self
         return $model;
     }
 
-    public function getWebhookUrl(): ?string
+    public function getWebhookUrls(): ?array
```
[the remark this answers is not in the document]
Comment (svenbw) [reply src/Model/Webhook/ShowResponse.php]: The reason why I kept the old functions (to the V2 interface) is to be able to check and remove the "legacy" webhooks; this matches the behaviour of the Mailgun control panel.
In future releases the legacy interfaces/functions should be removed from the library.

File: src/Model/Webhook/IndexResponse.php [hunk 2]
```diff
@@ -36,49 +44,92 @@ public static function create(array $data): self
...
         $data = $data['webhooks'] ?? $data;
 
-        $model->bounce = $data['bounce'] ?? [];
-        $model->deliver = $data['deliver'] ?? [];
-        $model->drop = $data['drop'] ?? [];
-        $model->spam = $data['spam'] ?? [];
-        $model->unsubscribe = $data['unsubscribe'] ?? [];
-        $model->click = $data['click'] ?? [];
-        $model->open = $data['open'] ?? [];
+        $model->legacy_bounce = $data['bounce'] ?? [];
```
[the remark this answers is not in the document]
Comment (svenbw) [reply src/Model/Webhook/IndexResponse.php]: Adding `urls` (and `url` for the legacy versions) is a good addition; shoudn't we return an empty array for the V3 functions (and not `null`) to indicate that there are no hooks linked?

**Comment by Nyholm** [comment]
This PR needs rebase

**Comment by spagr** [comment]
Please @svenbw, can You rebase this PR as as @Nyholm  requested above. I will be grateful for this, I also need to handle webhooks from API V3 ...

**Comment by svenbw** [comment]
Backwards compatibility fails because webhook urls are an array in the new API (for more info see [the API examples](https://documentation.mailgun.com/en/latest/api-webhooks.html#examples)).
Therefore the new method is called `getWebhookUrls` which returns an array.

A possible solution is to keep `getWebhookUrl` (next to the new call) but add an optional index as an argument. This function will return the n-th url in the array of url's. Default value for the index is 0, which would mimic the same behavior as before.

**Comment by DavidGarciaCat** [comment]
GIven getting green status for BC Checks is a requirement, I can't merge this update. Also, because it breaks the BC rule, it should be released as a new major version of the SDK.

@Nyholm thoughts?

**Comment by Nyholm** [comment]
Have a look at #749. 

I added a commit to make sure we dont break BC. I also did a rebase. I see that you've just merged in master into your branch. 

I could not (force) push to your fork since you made the PR from your fork's master.

**Comment by svenbw** [comment]
@Nyholm Your solution in the [Dont break BC commit](https://github.com/mailgun/mailgun-php/pull/749/commits/39cfa9586c1b2dbf5d1ee66eff87f785566d2591) is exactly what I suggest:

> Therefore the new method is called getWebhookUrls which returns an array.
> A possible solution is to keep `getWebhookUrl` (next to the new call) but add an optional index as an argument. This function will return the n-th url in the array of url's. Default value for the index is 0, which would mimic the same behavior as before.

I add the method `getWebhookUrls` and (optionally) modify the `getWebhookUrl` with the argument and default.

**Comment by Nyholm** [comment]
If you add that commit here and do a rebase, I can merge this PR instead of the other one

**Comment by DavidGarciaCat** [comment]
image

This one is flagged as failed.

Knowing that https://github.com/mailgun/mailgun-php/pull/749 is based on all the commits that @svenbw has pushed, and there is just this final fix, I guess it's faster if we merge the 749 and give credit to @svenbw for the update + @Nyholm for the last fix.

I will merge the code and release a new tagged version.

**Comment by DavidGarciaCat** [comment]
This code is now merged through https://github.com/mailgun/mailgun-php/pull/749 - Good job, guys!

**Pull request closed without merging**